src/components/task-board/TaskBoardColumn.jsx

```diff
@@ -1,27 +1,48 @@
-import React from "react";
+import React, { useEffect, useRef } from "react";
 import { Droppable } from "react-beautiful-dnd";
 import DarkModeContext from "../../services/theme-context";
 import BoardTask from "../global/Task/BoardTask";
 
 function TaskBoardColumn(props) {
-  const { column, tasks, subTasks, handleSortList } = props;
+  const {
+    column,
+    tasks,
+    subTasks,
+    sortListOpened,
+    onSortButtonClick,
+    handleSortList,
+  } = props;
+  const myRef = useRef();
+
+  /**
+   * handler for mouse click event, close the dropdown list when mouse clicked
+   * the sort list
+   * @param {*} e The mouse clicked event
+   */
+  const handleOutsideClick = (e) => {
```

Comment:
You should move this under all the hooks - this increases readability for future users

Reply:
This function is used inside the useEffect hook so I had to declare it before the useEffect hook, and inside the component. Maybe I can declare the function inside the useEffect hook instead?